core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -74,13 +74,12 @@ class SparkHadoopUtil extends Logging {
     }
   }
 
-  /**
-   * Return an appropriate (subclass) of Configuration. Creating config can initializes some Hadoop
-   * subsystems.
-   */
-  def newConfiguration(conf: SparkConf): Configuration = {
-    val hadoopConf = new Configuration()
 
+  /**
+    * Appends S3-specific, spark.hadoop.*, and spark.spark.buffer.size configurations to a Hadoop
```

Comment:
This is looking reasonable. If we're able to get a comment from @marmbrus about this particular aspect of the change (he put in the todo in https://github.com/apache/spark/commit/9aadcffabd226557174f3ff566927f873c71672e ) maybe that would confirm that this does need a special treatment and so this change makes sense.